osf/management/commands/backfill_domain_references.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+import re
+import logging
+import operator
+from functools import reduce
+
+from django.apps import apps
+from django.db.models import Q
+from django.core.management.base import BaseCommand
+from framework.celery_tasks import app as celery_app
+from osf.models import NotableDomain, DomainReference
+from django.contrib.contenttypes.models import ContentType
+
+logger = logging.getLogger(__name__)
+
+DOMAIN_MATCH_REGEX = re.compile(r'(?P<protocol>\w+://)?(?P<www>www\.)?(?P<domain>[\w-]+\.\w+)(?P<path>/\w*)?')
+DOMAIN_SEARCH_REGEX = r'(http://[^ \'}\[\]\~\(\)\/]+|https://[^ \'}\[\]\~\(\)\/]+)'
+
+
+@celery_app.task()
+def create_notable_domain_with_reference(domain, resource_id, resource_content_type_pk):
+    domain, created = NotableDomain.objects.get_or_create(
+        domain=domain,
+        defaults={'note': NotableDomain.Note.UNKNOWN}
+    )
+    DomainReference.objects.get_or_create(
```

Comment:
This is going to create DomainReferences to things that can't be Spammed (wikis and comments, notably), which will cause task failures elsewhere. Make sure that resources that aren't spammable are creating entries pointing to their parent resource.

Reply:
Comments are actually spammable and there's even some admin workflows that depend on this ability even though we rarely moderate comments anymore, but I do see what you mean. The wiki/comment is representing the whole node in this case, even though technically the GFK could point to any resource.